src/test/regress/sql/multi_router_planner.sql

```diff
@@ -0,0 +1,327 @@
+	FROM articles_hash
+	WHERE author_id = 1;
+
+-- below query hits a single shard, but it is not router plannable
+-- still router executable
+SELECT *
+	FROM articles_hash
+	WHERE author_id = 1 OR author_id = 17;
+
+-- below query hits two shards, not router plannable + not router executable
+SELECT *
+	FROM articles_hash
+	WHERE author_id = 1 OR author_id = 18;
+
+-- rename the output columns
+SELECT id as article_id, word_count * id as random_value
+	FROM articles_hash
+	WHERE author_id = 1;
+
+-- we can push down co-located joins to a single worker
+-- this is not router plannable but router executable
+SELECT a.author_id as first_author, b.word_count as second_word_count
+	FROM articles_hash a, articles_hash b
+	WHERE a.author_id = 10 and a.author_id = b.author_id
+	LIMIT 3;
+
+-- following join is neither router plannable, nor router executable
+SELECT a.author_id as first_author, b.word_count as second_word_count
+	FROM articles_hash a, articles_single_shard_hash b
+	WHERE a.author_id = 10 and a.author_id = b.author_id
+	LIMIT 3;
+
+-- single shard select with limit is router plannable
+SELECT *
+	FROM articles_hash
+	WHERE author_id = 1
+	LIMIT 3;
+
+-- single shard select with limit + offset is router plannable
+SELECT *
+	FROM articles_hash
+	WHERE author_id = 1
+	LIMIT 2
+	OFFSET 1;
+
+-- single shard select with limit + offset + order by is router plannable
+SELECT *
+	FROM articles_hash
+	WHERE author_id = 1
+	ORDER BY id desc
+	LIMIT 2
+	OFFSET 1;
+	
+-- single shard select with group by on non-partition column is router plannable
+SELECT id
+	FROM articles_hash
+	WHERE author_id = 1
+	GROUP BY id;
+
+-- single shard select with distinct is router plannable
+SELECT distinct id
+	FROM articles_hash
+	WHERE author_id = 1;
+
+-- single shard aggregate is router plannable
+SELECT avg(word_count)
+	FROM articles_hash
+	WHERE author_id = 2;
+
+-- max, min, sum, count are router plannable on single shard
+SELECT max(word_count) as max, min(word_count) as min,
+	   sum(word_count) as sum, count(word_count) as cnt
+	FROM articles_hash
+	WHERE author_id = 2;
+
+
+-- queries with aggregates and group by supported on single shard
+SELECT max(word_count)
+	FROM articles_hash
+	WHERE author_id = 1
+	GROUP BY author_id;
+
+SET client_min_messages to 'NOTICE';
+-- error out for queries with repartition jobs
+SELECT *
+	FROM articles_hash a, articles_hash b
+	WHERE a.id = b.id  AND a.author_id = 1;
+
+-- error out for queries which hit more than 1 shards
+SELECT *
+	FROM articles_hash
+	WHERE author_id >= 1 AND author_id <= 3;
+
+SET citus.task_executor_type TO 'real-time';
+
+-- Test various filtering options for router plannable check
+SET client_min_messages to 'DEBUG2';
+
+-- this is definitely single shard
+-- but not router plannable
+SELECT *
+	FROM articles_hash
+	WHERE author_id = 1 and author_id >= 1;
+
+-- not router plannable due to or
+SELECT *
+	FROM articles_hash
+	WHERE author_id = 1 or id = 1;
+	
+-- router plannable
+SELECT *
```

Comment:
It'd be nice to have a more complex qual expressions for both examples and counter-examples such as:
```SQL
* WHERE author_id = 1 and (id = 1 or id = 41) AND (word_count > 100 AND title != 'citus') ;
* WHERE author_id = 1 and (id = 1 or id = 41) AND (word_count > 100 OR title != 'citus') ;
* WHERE author_id = 1 and (id = 1 or id = 41) OR (word_count > 100 OR title != 'citus') ;
```

Also, let's have some tests where matching expression is on the middle/end:
 
```SQL
* WHERE (id = 1 or id = 41) AND (word_count > 100 AND title != 'citus') AND author_id = 1;
* WHERE (id = 1 or id = 41) AND author_id = 1 AND (word_count > 100 OR title != 'citus') ;
* WHERE (id = 1 or id = 41) OR  (author_id = 1 AND (word_count > 100 OR title != 'citus')) ;
```